src/main/java/com/aws/greengrass/clientdevices/auth/certificate/usecases/ConfigureCertificateAuthorityUseCase.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.certificate.usecases;
+
+import com.aws.greengrass.clientdevices.auth.CertificateManager;
+import com.aws.greengrass.clientdevices.auth.api.UseCases;
+import com.aws.greengrass.clientdevices.auth.configuration.CDAConfiguration;
+import com.aws.greengrass.clientdevices.auth.exception.CloudServiceInteractionException;
+import com.aws.greengrass.clientdevices.auth.exception.InvalidConfigurationException;
+import com.aws.greengrass.deployment.DeviceConfiguration;
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+import com.aws.greengrass.util.Coerce;
+
+import java.io.IOException;
+import java.security.KeyStoreException;
+import java.security.cert.CertificateEncodingException;
+import javax.inject.Inject;
+
+public class ConfigureCertificateAuthorityUseCase implements UseCases.UseCase<Void, CDAConfiguration> {
+    private final CertificateManager certificateManager;
+    private final DeviceConfiguration deviceConfiguration;
+    private static final Logger logger = LogManager.getLogger(ConfigureCertificateAuthorityUseCase.class);
+
+
+    @Inject
+    public ConfigureCertificateAuthorityUseCase(
+            CertificateManager certificateManager,
+            DeviceConfiguration deviceConfiguration) {
+        this.certificateManager = certificateManager;
+        this.deviceConfiguration = deviceConfiguration;
+    }
+
+    @Override
+    public Void execute(CDAConfiguration configuration) throws InvalidConfigurationException,
+            CertificateEncodingException, IOException, KeyStoreException {
+        // NOTE: This is not the final shape of this useCase we are just taking the logic out from
+        //  the ClientDeviceAuthService first.
+        String thingName = Coerce.toString(deviceConfiguration.getThingName());
+
+        try {
```

Comment:
This will be further expanded and the certificate manager will disappear. This PR just moves the logic that was on the ClientDeviceAuthService to this useCase